resources/_splunk_app_upgrade.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+# Cookbook Name:: cerner_splunk_ingredient
+# Resource:: _splunk_app_upgrade
+#
+# Resource for installing and configuring Splunk apps
+
+class SplunkAppUpgrade < Chef::Resource
+  include CernerSplunk::PlatformHelpers, CernerSplunk::PathHelpers, CernerSplunk::ResourceHelpers
+
+  property :name, String, name_property: true, identity: true
+  property :install_dir, String, required: true
+  property :version, [String, CernerSplunk::SplunkVersion], required: true
+
+  resource_name :_splunk_app_upgrade
+
+  default_action :nothing
+
+  def after_created
+    version CernerSplunk::SplunkVersion.from_string(version) if version
```

Comment:
So an odd thing here... Property version is documented of taking a String or a SplunkVersion... so if I as a consumer pass a SplunkVersion, this would actually fail...